### PR TITLE
Fix build errors found with Serac's configuration

### PR DESCRIPTION
**Pull request by white238**
I swear I tested this. I am not sure how the hdf5 ever worked. Serac is the only configuration I know that turns on mpi for hdf5 but I turned it on 25 days ago and there has been multiple TPL builds since then...

### CODE REVIEW

File: src/thirdparty/tests/CMakeLists.txt [hunk 1]
```diff
@@ -59,10 +56,15 @@ endif()
...
+
+    set( hdf5_smoke_dependencies hdf5 gtest )
+
+    blt_list_append( TO hdf5_smoke_dependencies ELEMENTS mpi IF ENABLE_MPI)
+
     blt_add_executable(NAME hdf5_smoke_test
                        SOURCES hdf5_smoke.cpp
                        OUTPUT_DIR ${TEST_OUTPUT_DIRECTORY}
-                       DEPENDS_ON hdf5 gtest
+                       DEPENDS_ON ${hdf5_smoke_dependencies}
```
Comment (gzagaris) [inline src/thirdparty/tests/CMakeLists.txt:67]: I guess you are building with an MPI version of HDF5....

**Review by gzagaris — APPROVED** [review]

File: src/thirdparty/tests/CMakeLists.txt (hunk 1 above)
Comment (white238) [reply src/thirdparty/tests/CMakeLists.txt:67]: ... but how did this not break 25 days ago... maybe spack has tests off by default.

Comment (gzagaris) [reply src/thirdparty/tests/CMakeLists.txt:67]: Hmm...were you using a different version of CMake perhaps? I have run into an issue where different versions of CMake wouldn't propagate dependencies correctly...

**Review by agcapps — APPROVED** [review]
Thumbs up

**Pull request merged**